kafka/consumer.go

```diff
@@ -0,0 +1,72 @@
+// Package kafka provides producer and consumer to work with kafka topics
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/Shopify/sarama"
+	"github.com/lvl484/positioning-filter/matcher"
+)
+
+type Consumer struct {
+	ConsumerGroup sarama.ConsumerGroup
+	Config        *Config
+	closeChan     chan bool
+}
+
+func NewConsumer(config *Config) (*Consumer, error) {
+	addr := []string{fmt.Sprintf("%v:%v", config.Host, config.Port)}
+	version, err := sarama.ParseKafkaVersion(config.Version)
+
+	if err != nil {
+		return nil, err
+	}
+
+	saramaConfig := sarama.NewConfig()
+	saramaConfig.Version = version
+	saramaConfig.Consumer.Return.Errors = true
+
+	consumerGroup, err := sarama.NewConsumerGroup(addr, config.ConsumerGroupID, saramaConfig)
+
+	if err != nil {
+		return nil, err
+	}
+
+	closeChan := make(chan bool)
+
+	return &Consumer{
+		ConsumerGroup: consumerGroup,
+		Config:        config,
+		closeChan:     closeChan,
+	}, nil
+}
+
+func (c *Consumer) Consume(matcher matcher.Matcher, producer Producer) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	handler := newConsumerGroupHandler(matcher, producer)
+
+	for {
+		select {
+		case _, ok := <-c.closeChan:
+			if !ok {
```

Comment:
Please check this case, start consuming and send termination signal

Reply:
func just will be closed